### DISCUSSION TITLE
Schematic Selection Centering Option

### DISCUSSION THREAD

**Discussion by VinceSMC** [BG-Software-LLC/SuperiorSkyblock2#2126] [Pending Feature Requests]
Hey Ome_R,

I'm working on a new Skyblock server setup for SMC Developments and was wondering if you could add a small but powerful feature to SuperiorSkyblock. I've seen a few people ask for it already, and I'm willing to pay to get it done.

Basically, it's about adding a new "centering" type for the island schematic selection within the island border. Right now, it's based on the player spawn point, but I'd like an option to base it on the schematic selection itself. This way, using something like a double island won't go outside the border:

![2024-06-30_12 05 18](https://github.com/BG-Software-LLC/SuperiorSkyblock2/assets/97965361/a5de1dd2-c3b6-48ed-a313-6f951d58b5d6)

It could be something like an additional option in the island-creation.yml menu for schematic items:
`center-type: spawn-point/schematic-selection`

I think this would be super handy and solve some issues with schematics going outside the border. We're hoping to launch soon, so it would be awesome if this could be prioritized.

Thanks for considering it!

**Comment by OmerBenGera** [comment]
Check out latest dev build
You can add `spawn-offset` to the schematic in the creation menu, which determines the offset to teleport you from the center of the schematic. The schematic will still be placed according to where you saved it, but you can teleport yourself to another place.
Note that this place must be 'safe', otherwise the plugin will try and look for a new place for the teleportation location.

<https://wiki.bg-software.com/superiorskyblock/overview/schematics#adding-your-own-schematic>

**Comment by VinceSMC** [comment]
**Thanks a lot!** The new spawn-offset feature is working perfectly. I really appreciate the swift implementation.
